### PR TITLE
chore(react-avatar): Adding unit and vr tests for AvatarGroup

### PR DESCRIPTION
**Pull request by sopranopillow**
## Current Behavior

AvatarGroup has no unit and visual regression tests.

## New Behavior

AvatarGroup has unit and visual regression tests.

## Related Issue(s)

#22240

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/82337ca348673485e78b9072431c00c222cd985a'>82337ca348673485e78b9072431c00c222cd985a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/Avatar-mount_base.html">1622</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/Avatar-mount.html">1631</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/Button-mount_base.html">1283</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/Button-mount.html">1208</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProvider-mount_base.html">2843</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProvider-mount.html">2798</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProviderWithTheme-mount_base.html">953</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProviderWithTheme-mount.html">916</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">962</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">855</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">953</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">973</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/MakeStyles-mount_base.html">2515</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23549/perf-test-react-components/MakeStyles-mount.html">2513</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx [hunk 1]
```diff
@@ -2,24 +2,133 @@ import * as React from 'react';
...
 
 describe('AvatarGroup', () => {
-  // TODO: Remove component-has-static-classnames-object from disabled tests.
   isConformant({
     Component: AvatarGroup,
     displayName: 'AvatarGroup',
     disabledTests: [
       'component-has-static-classname',
       'component-has-static-classname-exported',
+      // AvatarGroup's overflowContent is rendered in a portal, therefore this test won't work
```
Comment (spmonahan) [inline packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx:14]: You can test this but you have to override some functions for the conformance tests. I have an issue to make this better but I haven't had time to look into a general purpose solution.

In the mean time, [Tooltip is an example of how to conformance test Portals](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-tooltip/src/components/Tooltip/Tooltip.test.tsx#L8-L28)

File: packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx [hunk 2]
```diff
@@ -2,24 +2,133 @@ import * as React from 'react';
...
+        <AvatarGroupItem name="Daisy Phillips" />
+        <AvatarGroupItem name="Robert Tolbert" />
+        <AvatarGroupItem name="Kevin Sturgis" />
+      </AvatarGroup>,
+    );
+
+    expect(screen.getByRole('button').textContent).toBe('');
+  });
+
+  it('ignores maxAvatars when using ap pie layout', () => {
```
Comment (spmonahan) [inline packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx:55]:
```suggestion
  it('ignores maxAvatars when using pie layout', () => {
```

File: apps/vr-tests-react-components/src/stories/AvatarGroup.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  })
+  .addStory('layoutPie', () => <AvatarGroupList layout="pie" />)
+  .addStory('layoutStack', () => <AvatarGroupList layout="stack" />)
+  .addStory('overflowIndicator', () => <AvatarGroupList overflowIndicator="icon" />);
+
+// Interactive stories
+storiesOf('AvatarGroup Converged', module)
+  .addDecorator(TestWrapperDecorator)
+  .addDecorator(story => (
+    <Screener steps={new Screener.Steps().click('#show-overflow').snapshot('overflowConentOpen').end()}>
```
Comment (khmakoto) [inline apps/vr-tests-react-components/src/stories/AvatarGroup.stories.tsx:74]:
```suggestion
    <Screener steps={new Screener.Steps().click('#show-overflow').snapshot('overflowContentOpen').end()}>
```

File: change/@fluentui-react-avatar-0ebe3e03-9d8f-48f7-a4c0-2a95b7014e2e.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "chore: Adding unit tests for AvatarGroup. ",
+  "packageName": "@fluentui/react-avatar",
+  "email": "esteban.230@hotmail.com",
+  "dependentChangeType": "patch"
+}
```
Comment (khmakoto) [inline change/@fluentui-react-avatar-0ebe3e03-9d8f-48f7-a4c0-2a95b7014e2e.json:7]: This can be of type none since it is just a change in tests

```suggestion
{
  "type": "none",
  "comment": "chore: Adding unit tests for AvatarGroup. ",
  "packageName": "@fluentui/react-avatar",
  "email": "esteban.230@hotmail.com",
  "dependentChangeType": "none"
}
```

File: packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx (hunk 1 above)
Comment (sopranopillow) [reply packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx:14]: @spmonahan I think for this case, the test should be disabled. I was able to get the custom function working, but I think that's not what we should be doing. When adding the getAvatarGroupElement target function we are saying "test isConformant but first get this element and test against the result". The problem with this is that we are then focusing on the PopoverSurface. This is why I disabled it since I think it's more worthwhile that we test the root and ref rather than just test the overflow content which doesn't handle many of the other tests.

One thing I would propose would be to add a second isConformat test that will only test the overflowContent part of AvatarGroup and leave the isConformant above as is. Seems hacky, but it would cover all component. 

What are your thoughts on this?

On a side note, this brings then another case that we should consider in our conformance tests.

Comment (spmonahan) [reply packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.test.tsx:14]: I'm fine with disabling in this case. I'm not sure the best way to generally handle conformance for situations like this but I also haven't had time to give it much thought. It's looking like we'll have a few different components that make use of Portal by the time I get back to it so hopefully we'll be able to devise a generic solution.

**Review by khmakoto — APPROVED** [review]
